src/components/DocExplorer.js

```diff
@@ -148,6 +181,142 @@ export class DocExplorer extends React.Component {
   }
 }
 
+// Render Search Results
+class SearchDoc extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    return (this.props.schema !== nextProps.schema) ||
+      (this.props.searchValue !== nextProps.searchValue);
+  }
+
+  render() {
+    const searchValue = this.props.searchValue;
+    const schema = this.props.schema;
+    const onClickType = this.props.onClickType;
+    const onClickField = this.props.onClickField;
+
+    const typeMap = schema.getTypeMap();
+
+    // Compose a search index for all possible arguments from every types.
+    let possibleArguments = [];
```

Comment:
This isn't used anywhere below, maybe we should remove it for this PR.

Also, we don't want to compute this index on every render since it's likely to be expensive for larger schema. We only want to compute an index once per schema.